provider/pkg/provider/util.go

```diff
@@ -139,6 +139,11 @@ func getActiveClusterFromConfig(config *clientapi.Config, overrides resource.Pro
 // pruneMap builds a pruned map by recursively copying elements from the source map that have a matching key in the
 // target map. This is useful as a preprocessing step for live resource state before comparing it to program inputs.
 func pruneMap(source, target map[string]any) map[string]any {
+	// If either map is nil, return nil.
```

Comment:
Changes look good overall, but it would be good to double-check the unit tests for these functions and make sure the updated logic is adequately covered there.

Reply:
Existing unit tests passed. Have expanded on this in a follow-up commit.